File: 647-palindromic-substrings/palindromic-substrings.py
class Solution:
    def countSubstrings(self, s: str) -> int:

        def countPalindromes(s, l, r):
            count = 0
            while l >= 0 and r < len(s) and s[l] == s[r]:
                count += 1
                l -= 1
                r += 1
            return count
        
        total_count = 0

        for i in range(len(s)):
            
            # for odd count ->
            total_count += countPalindromes(s, i, i)

            # for even count ->
            total_count += countPalindromes(s, i, i+1)

        return total_count
        

File: 647-palindromic-substrings/palindromic-substrings.java
class Solution {

    public int countPalindrome(String s, int start, int end) {

        int c = 0;
        while (start >= 0 && end < s.length() && s.charAt(start) == s.charAt(end)) {
            c += 1;
            start -= 1;
            end += 1;
        }
        return c;
    }

    public int countSubstrings(String s) {

        int count = 0;
        for (int i = 0; i < s.length(); i++) {
            
            count += countPalindrome(s, i, i);
            count += countPalindrome(s, i, i + 1);
        }
        return count;
    }
}